src/httpClient/httpClient.ts

```diff
@@ -60,29 +52,22 @@ export default class HttpClient {
     this.enableRetry = options?.enableRetry ?? false;
     this.timeout = options?.timeout;
     this.clientExceptionStatusCodeMapOverride = options?.clientExceptionStatusCodeMapOverride;
-    this.client =
-      options?.client ??
-      axios.create({
-        adapter: (() => {
-          let adapters = axios.defaults.adapter as AxiosAdapter;
-          if (this.enableCache) {
-            const cache = setupCache({
-              maxAge: 5 * 60 * 1000, // all items are cached for 5 minutes
-              readHeaders: false, // ignore cache control headers in favor of the static 5 minutes
-              readOnError: true,
-              exclude: {
-                query: false, // also cache requests with query parameters
-              },
-              ...options?.cacheOptions, // allow to overwrite the defaults except of cache-key
-              key: (req) => HttpClient.generateCacheKey(req),
-            });
-
-            // debounce concurrent calls with the same cacheKey so that only one HTTP request is made
-            adapters = createDebounceRequestAdapter(cache.adapter, HttpClient.generateCacheKey);
-          }
-          return adapters;
-        })(),
+    this.client = options?.client ?? axios.create();
+
+    if (this.enableCache) {
+      setupCache(this.client, {
+        // 5 minutes TTL
+        ttl: 5 * 60 * 1000,
+        // Ignore cache-control headers in favor of static TTL
+        interpretHeader: false,
+        // Serve stale cache on error
+        staleIfError: true,
+        // Use custom cache key to include auth, url, params and body hash
+        generateKey: (req) => HttpClient.generateCacheKey(req as AxiosRequestConfig),
+        // Allow overriding defaults via provided options
+        ...options?.cacheOptions,
```

Comment:
Does the new adapter support using Redis? We depend on the `RedisDefaultStore` from the old adapter https://github.com/lisaogren/axios-cache-adapter/blob/master/src/redis-default.js